pkg/operator/ceph/nfs/controller.go

```diff
@@ -382,3 +411,30 @@ func updateStatus(observedGeneration int64, client client.Client, name types.Nam
 	}
 	logger.Debugf("nfs %q status updated to %q", name, status)
 }
+
+func (r *ReconcileCephNFS) updateNFSCephxStatus(namespacedName types.NamespacedName, cephxStatus cephv1.CephxStatus) error {
```

Comment:
There is already an `updateStatus()` function. Let's use that as the only status update method so we don't have 2. That will be confusing for future maintenance.

I think this is a good reference to see how the old status method was modified for another component: https://github.com/rook/rook/pull/16313/files#diff-481374f7e18758377bba79e9d09a26b97484050e4bb66a7a68202477f68f2141R299